lib/Agrammon/DB/Dataset.pm6

```diff
@@ -194,14 +194,43 @@ class Agrammon::DB::Dataset does Agrammon::DB {
 
         my $ds = self!create-dataset( $new-dataset, $new-username, $!version, $!model);
         self.with-db: -> $db {
+            # clone inputs
             $db.query(q:to/SQL/, $ds<id>, $old-username, $old-dataset);
                 INSERT INTO data_new (data_dataset, data_var, data_instance, data_val, data_instance_order, data_comment)
                      SELECT $1, data_var, data_instance, data_val, data_instance_order, data_comment
                        FROM data_new
                       WHERE data_dataset = dataset_name2id($2, $3)
             SQL
 
+            # get branched inputs from new dataset
+            my @rows = $db.query(q:to/SQL/, $old-username, $new-dataset).arrays;
```

Comment:
If it's from the new dataset, should it be `$new-username`?

Reply:
That's true, I forgot about the case where we copy the dataset into the account of another user.
In this case, however, it can actually be even simpler by using `$ds<id>`. Fixed.